### PR TITLE
[subresource loading] Specify how UA parses JSON (fix #728)

### LINKED ISSUE #728: [subresource-loading] Specify how UA parses JSON in <script type=webbundle>
Meta issue: #708.

We have to specify how UA parses JSON in `<script type=webbundle>`.

### PR DESCRIPTION
**Pull request by hayatoito**
Specify how UA parses JSON.

***
<a href="https://pr-preview.s3.amazonaws.com/hayatoito/webpackage/pull/733.html" title="Last updated on Apr 21, 2022, 3:34 AM UTC (e4761b0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/webpackage/733/fcb50a7...hayatoito:e4761b0.html" title="Last updated on Apr 21, 2022, 3:34 AM UTC (e4761b0)">Diff</a>

### CODE REVIEW

File: subresource-loading.bs [hunk 1]
```diff
@@ -438,15 +438,55 @@ detailed motivation.
...
-1. If |parsed| is not a [=map=], then throw a {{TypeError}} indicating that the
-   top-level value needs to be a JSON object.
-1. Create a [=web bundle parse result=] from |parsed| and |baseURL|, and return
-   the [=web bundle parse result=].
-
-ISSUE: Specify the details. Throw a {{TypeError}} etc. on a parse error.
-Relative URLs are resolved on |baseURL|.
+1.  Let |parsed| be the result of [=parse JSON into Infra values|parsing JSON
+    into Infra values=] given |sourceText|.
+1.  If |parsed| is not a [=map=], then throw a {{TypeError}} indicating that the
```
Comment (irori) [inline subresource-loading.bs:443]: This is different from current Chromium implementation which throws a SyntaxError, but I think this makes more sense. We should fix our implementation.

File: subresource-loading.bs [hunk 2]
```diff
@@ -438,15 +438,55 @@ detailed motivation.
...
+    top-level value needs to be a JSON object.
+1.  If |parsed|["`source`"] does not [=map/exist=], then throw a {{TypeError}}.
+1.  If |parsed|["`source`"] is not a [=string=], then throw a {{TypeError}}.
+1.  Let |source| be the result of [=URL parser|parsing=] |parsed|["`source`"]
+    with |baseURL| as the base URL.
+1.  If |source| is null, then throw a {{TypeError}}.
+1.  Let |credentials| be "`same-origin`".
+1.  If |parsed|["`credentials`"] [=map/exists=], then:
+    1. If |parsed|["`credentials`"] is "`omit`", then set |credentials| to
```
Comment (irori) [inline subresource-loading.bs:452]: Just to confirm, this is case-sensitive comparison (i.e. https://infra.spec.whatwg.org/#string-is), correct?

File: subresource-loading.bs (hunk 1 above)
Comment (hayatoito) [reply subresource-loading.bs:443]: Ack.

File: subresource-loading.bs (hunk 2 above)
Comment (hayatoito) [reply subresource-loading.bs:452]: Yeah, case-sensitive comparison is intentional. Thanks for the link for "string-is". I didn't know "string-is" is defined.

**Review by irori — APPROVED** [review]
lgtm

**Pull request merged**